### ISSUE TITLE
[derive] [wish] Support mutual inductive types

### ISSUE THREAD

**Issue by Columbus240** [LPCIC/coq-elpi#261]
Would be a natural extension (from the user’s POV) to the current functionality. Though it probably is a big project.

**Comment by gares** [comment]
It is doable, but it is a lot of work for a class of examples which never looked very convincing to me. Maybe you have a good one to share?

**Comment by Columbus240** [comment]
No, currently not. I had a mutually-inductive definition, when I wrote the issue, because Coq didn't allow how I used `Forall`. This is about the only situation, where I come across mutually-inductive types.
But I just rewrote the definition and now Coq accepts it with `Forall`.
There is some discussion on this [StackExchange](https://cs.stackexchange.com/questions/136658/is-mutual-inductive-type-definition-essential-in-coq-core-language) about dealing with mutual inductives, but no interesting example.

**Comment by Columbus240** [comment]
[Previous discussion on Zulip](https://coq.zulipchat.com/#narrow/stream/253928-Elpi-users.20.26.20devs/topic/Support.20for.20mutually.20inductive.20types/near/217694418), November 2020
